Add ColorWheel widget for color picking

mrzv opened the pull request:
The drawing commands are taken directly from the NanoVG demo.

wjakob commented:
Hi,

that looks cool, though there are still some things I'd like to see fixed before merging.

Right now, the example application looks really ugly after the change :) -- partly because the color wheel is so huge, and it messes up the layout for everything else.

When fixing the wheel to a smaller size (e.g. 100x100), the proportions don't seem right anymore.

Also, it would be best if the callback takes the updated color value as an argument like in some other widgets (e.g. Slider).

Thanks,
Wenzel

mrzv commented:
I'll play around with the sizes. To be honest, currently the relationship between preferred layout, fixed layout, and the actual layout are a little mysterious to me.

I originally gave the color value to the callback, but then I changed it to the current way. The reason is that I didn't like spelling out `Vector3f`, and it's quite easy to capture the color wheel in the lambda and just call `color()`.

BTW, I'd love to be have a color picker button (that's colored the way of the chosen color) that when pressed opens up a color wheel. Currently, I just open up a Popup and put the color wheel into it. But if you have the energy to make it so that one could open the color wheel by itself (without a popup underneath), that would  be nice. Currently, it's outside my comfort level with nanogui.

wjakob commented:
Awesome, thanks. I made a few more minor changes related to formatting and event handling (in the next commit).
